src/firefly/java/edu/caltech/ipac/firefly/server/query/ObsCorePackager.java

```diff
@@ -155,15 +156,12 @@ public static List<FileInfo> parseDatalink(URL url, String prepend_file_name, St
                         ext_file_name = makeDataLinkFolder ? prepend_file_name + "/" + ext_file_name : ext_file_name;
                     }
 
-                    if (products != null) {
-                        // Check if products exist and contains sem
+                    if (products != null) { // Check if products exist and contains sem
```

Comment:
@robyww @loitly I left this as a `null` check, because I add a check for null & empty for the string `productTypes` at the beginning on `computeFileGroup` function above.